lua/hover/config.lua

```diff
@@ -1,14 +1,18 @@
 local M = {}
 
+---@class HoverConfig
```

Comment:
Figured I could add some more type annotations, happy to remove these (incl. the `@async` one) if you'd like